es/lib/isEmpty.js

```diff
@@ -1,10 +1,38 @@
+function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }
+
 import assertString from './util/assertString';
 import merge from './util/merge';
 var default_is_empty_options = {
-  ignore_whitespace: false
+  ignore_whitespace: false,
+  to_string: false
 };
 export default function isEmpty(str, options) {
-  assertString(str);
+  var emptyValues = [undefined, null, [], false, 0, '', '0'];
   options = merge(options, default_is_empty_options);
-  return (options.ignore_whitespace ? str.trim().length : str.length) === 0;
+
+  if (options.to_string) {
```

Comment:
Not really sure about the "cleanliness" of this approach vs. the succinct code we had earlier...

Reply:
The current approach will make sure to check every possible value that results into an empty value not only strings.